web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/TraceContextCorrelation.java

```diff
@@ -0,0 +1,288 @@
+
+        try {
+            if (request == null) {
+                InternalLogger.INSTANCE.error("Failed to resolve correlation. request is null.");
+                return;
+            }
+
+            if (response == null) {
+                InternalLogger.INSTANCE.error("Failed to resolve correlation. response is null.");
+                return;
+            }
+
+            if (requestTelemetry == null) {
+                InternalLogger.INSTANCE.error("Failed to resolve correlation. requestTelemetry is null.");
+                return;
+            }
+
+
+            String traceId = request.getHeader(CORRELATION_HEADER_NAME);
+
+            Traceparent outGoingTraceParent = null;
+
+            if (traceId == null || traceId.isEmpty()) {
+                outGoingTraceParent = new Traceparent();
+
+                // represents the id of the current request.
+                requestTelemetry.setId(outGoingTraceParent.getTraceId() + "-" + outGoingTraceParent.getSpanId());
+
+                // represents the trace-id of this distributed trace
+                requestTelemetry.getContext().getOperation().setId(outGoingTraceParent.getTraceId());
+
+                // set parentId as null because this is the is the originating request
+                requestTelemetry.getContext().getOperation().setParentId(null);
+            } else {
+                Traceparent incomingTraceparent = Traceparent.fromString(traceId);
+
+                // create outgoing traceParent using the incoming header
+                outGoingTraceParent = new Traceparent(0, incomingTraceparent.getTraceId(),
+                    null, incomingTraceparent.getTraceFlags());
+
+                // set id of this request
+                requestTelemetry.setId(outGoingTraceParent.getTraceId() + "-" + outGoingTraceParent.getSpanId());
+
+                // represents the trace-id of this distributed trace
+                requestTelemetry.getContext().getOperation().setId(outGoingTraceParent.getTraceId());
+
+                // represents the parent-id of this request which is combination of traceId and incoming spanId
+                requestTelemetry.getContext().getOperation().setParentId(outGoingTraceParent.getTraceId() + "-" +
+                    incomingTraceparent.getSpanId());
+
+            }
+
+            String tracestate = request.getHeader(CORRELATION_COMPANION_HEADER_NAME);
+            Tracestate tracestateObject = Tracestate.fromString(tracestate);
+            Map<String, String> tracestatePropertiesMap = getPropertiesMap(tracestateObject);
+
+            // create outbound tracestate to be propagated to downstream calls
+            Tracestate outboundTracestate = createOutboundTracestate(tracestatePropertiesMap, getAppId());
+            ThreadContext.getRequestTelemetryContext().setTracestate(outboundTracestate);
+
+            // Let the callee know the caller's AppId
+            addTracestateInResponseHeader(response);
```

Comment:
we should not set tracestate in response header

Reply:
@lmolkova, what should we pass in the response header to achieve the same functionality? Should we just use the previous header? And why not tracestate?

Reply:
W3C does not define response headers. we should continue responding with Request-Context to everything

Reply:
That makes sense. I will update this one. The question comes in mind then what happens when one wants to use OpenCensus SDKs and just change the end-point to go to AI endpoint. This wouldn't work definitely. Do we want to introduce this legacy in the brand new implementation. 
I am aware .NET has done this! @reyang what are your thoughts here? 
PS : The work is minimal, but it's like once we ship something it's difficult to revert back.